#!/usr/bin/env python
PACKAGE = "laser_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Activate",	            bool_t, 		0,  "Activate GUI",		                    False)
gen.add("H_Min",		        double_t,		0,	"Minimum amount of red required",       0.1,    0.0,    5.0)
gen.add("S_Min",	            double_t, 		0,	"Minimum amount of green required",     0.1,    0.0,    5.0)
gen.add("V_Min",		        double_t,		0,	"Minumum amount of blue required",      0.1,    0.0,    5.0)
gen.add("H_Max",                double_t,       0,  "Maximum amount of red required",       0.9,    0.0,    5.0)
gen.add("S_Max",                double_t,       0,  "Maximum amount of green required",     1.9,    0.0,    5.0)
gen.add("V_Max",                double_t,       0,  "Maximum amount of blue required",      1.9,    0.0,    5.0)
gen.add("Sensitivity",          int_t,          0,  "Sensitivity value (for shape node)",   20,     0,      40)
gen.add("Blur",                 int_t,          0,  "Blur value (for shape node)",          10,     0,      40)


exit(gen.generate(PACKAGE, "laser_detection", "HsxParams"))
